avaframe/in2Trans/shpConversion.py

```diff
@@ -93,72 +93,79 @@ def SHP2Array(infile, defname=None):
     Coordz = np.empty((0))
     start = 0
     nParts = []
+    nAdded = []
 
     for n, item in enumerate(shps):
         pts = item.points
-        # is there a z coordinate?
-        try:
-            # for dams
-            zs = item.z
-        except AttributeError:
-            zs = [0.0] * len(pts)
-        # add info on number of parts
-        nParts.append(list(item.parts) + [len(item.points)])
-
-        # check if records are available and extract
-        if records:
-            # loop through fields
-            for (name, typ, size, deci), value in zip(sf.fields[1:], records[n].record):
-                # get entity name
-                name = name.lower()
-                if (name == 'name'):
-                    layername = str(value)
-                if (name == 'thickness') or (name == 'd0'):
-                    thickness = value
-                if (name == 'ci95'):
-                    ci95 = value
-                if (name == 'slope'):
-                    # for dams
-                    slope = value
-                if (name == 'rho'):
-                    rho = value
-                if (name == 'sks'):
-                    sks = value
-                if (name == 'iso'):
-                    iso = value
-                if (name == 'layer'):
-                    layerN = value
-            # if name is still empty go through file again and take Layer instead
-            if ((type(layername) is bytes) or (layername is None)):
+        if len(pts) > 0:
```

Comment:
Maybe I'm thinking too simple, but could we just to a 
if len <= 0:
  continue? 
And don't do all the other bookkeeping? 
I.e. if an item without points in encountered, we just don't do anything? I'm pretty sure the idlist could be built inside the first loop. 


If this is not possible for some reason, then I suggest trying to split the nesting...

Reply:
use continue and loop also over sf.records() in first loop to access records oid for idList